### PR TITLE
MPI changes

### PR DESCRIPTION
**Pull request by mand35**
simplifications, load imbalance description and MPI commands examples

### CODE REVIEW

**Comment by chrisdjscott** [comment]
@mand35 looks like this has conflicts now that the changes from Alex and Wolfgang were merged - do you want to update the pr or were your changes incorporated in the pr Alex and Wolfgang put in?

**Comment by mand35** [comment]
merged and cleaned

File: _lessons/python-scatter/05-mpi.md [hunk 1]
```diff
@@ -36,6 +31,7 @@ As the process runs, the program may need to exchange data ("messages" - hence t
  * there are no serial sections in MPI code and hence MPI programs need to be written to run in parallel from the beginning to the end
  * some algorithms are difficult to implement efficiently with MPI due its distributed memory approach and communication overheads
  * it is easy to create "dead locks" where one or more processes get stuck waiting for a message
+ * specified computing resources are occupied the whole run time, no dynamic allocation/deallocation of CPU cores
```
Comment (chrisdjscott) [inline _lessons/python-scatter/05-mpi.md]:
```suggestion
 * the requested computing resources are occupied the whole run time, i.e. there is no dynamic allocation/deallocation of CPU cores
```

File: _lessons/python-scatter/05-mpi.md [hunk 2]
```diff
@@ -148,7 +148,10 @@ mpiexec -n 8 python scatter.py
 
 ## How to use MPI to accelerate `scatter.py`
 
- * At the top: `from mpi4py import MPI`. This will initialise MPI. The number of processes is `nprocs`.
+ * `from mpi4py import MPI` at the top. This will initialise MPI.
+ * `comm = MPI_COMM_WORLD` sets the communicator.
```
Comment (chrisdjscott) [inline _lessons/python-scatter/05-mpi.md]:
```suggestion
 * `comm = MPI.COMM_WORLD` gets the communicator.
```

File: _lessons/python-scatter/05-mpi.md [hunk 3]
```diff
@@ -148,7 +148,10 @@ mpiexec -n 8 python scatter.py
...
+ * `nprocs = comm.Get_size()` stores the total number of processes.
```
Comment (chrisdjscott) [inline _lessons/python-scatter/05-mpi.md]:
```suggestion
 * `nprocs = comm.Get_size()` gets the total number of processes.
```

File: _lessons/python-scatter/05-mpi.md [hunk 4]
```diff
@@ -148,7 +148,10 @@ mpiexec -n 8 python scatter.py
...
+ * `pe = comm.Get_rank()` stores the actual rank of an process.
```
Comment (chrisdjscott) [inline _lessons/python-scatter/05-mpi.md]:
```suggestion
 * `pe = comm.Get_rank()` gets the actual rank of a process.
```

**Review by chrisdjscott — CHANGES_REQUESTED** [review]
@mand35 looks good, just couple of small changes and 1 fix

File: _lessons/python-scatter/05-mpi.md (hunk 4 above)
Comment (mand35) [reply _lessons/python-scatter/05-mpi.md]: these lines get the comm/rank/... and store it an variable. So one could argue, but I am fine with the change.

**Review by chrisdjscott — APPROVED** [review]

**Pull request merged**